bsp/src/org/jetbrains/bsp/project/test/BspTestConfigurationForm.java

```diff
@@ -0,0 +1,167 @@
+package org.jetbrains.bsp.project.test;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.ui.CollectionListModel;
+import com.intellij.ui.DocumentAdapter;
+import com.intellij.ui.components.JBList;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.uiDesigner.core.GridConstraints;
+import com.intellij.uiDesigner.core.GridLayoutManager;
+import com.intellij.uiDesigner.core.Spacer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.bsp.data.BspMetadata;
+import org.jetbrains.plugins.scala.util.JListCompatibility;
+import org.jetbrains.plugins.scala.util.JListCompatibility.CollectionListModelWrapper;
+import scala.collection.JavaConverters;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import java.awt.*;
+import java.awt.event.ItemEvent;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+
+public class BspTestConfigurationForm {
+	final Project project;
+	JPanel mainPanel;
+	JComboBox testModeCombobox;
+
+	JPanel testClassFormWrapper;
+	JTextField testClassNameRegex;
+	JBList matchedClassesList;
+	CollectionListModelWrapper matchedClassesModel = new CollectionListModelWrapper(new CollectionListModel<String>(Collections.emptyList()));
+
+	@SuppressWarnings("unchecked")
+	public BspTestConfigurationForm(Project project) {
+		this.project = project;
+		$$$setupUI$$$();
+		{ // init combo box
+			for (TestMode sm : TestMode.values())
+				testModeCombobox.addItem(sm);
+			testModeCombobox.addItemListener(e -> {
+				if (e.getStateChange() == ItemEvent.SELECTED)
+					onSelectModeChanged((TestMode) e.getItem());
+			});
+		}
+		{ // init matched classes list
+			JListCompatibility.setModel(matchedClassesList, matchedClassesModel.getModelRaw());
+			updateMatchedClassesList(Collections.emptyList());
+			testClassNameRegex.getDocument().addDocumentListener(new DocumentAdapter() {
+				@Override
+				protected void textChanged(@NotNull DocumentEvent e) {
+					onClassNameRegexChanged();
+				}
+			});
+		}
+	}
+
+	private void onClassNameRegexChanged() {
```

Comment:
This is invoked on every keypress and triggers a regular expression match. Is it a good idea ? I could not find a better action listener, which will notify on more reasonable frequency.

Reply:
Only on Keypresses in class name? I think that's rare, so not a big deal.

Reply:
yeah, but the worrying thing is that it triggers a regex compilation and search